src/github.com/matrix-org/dendrite/roomserver/input/consumer.go

```diff
@@ -87,7 +90,14 @@ func (c *Consumer) Start() error {
 func (c *Consumer) consumePartition(pc sarama.PartitionConsumer) {
 	defer pc.Close()
 	for message := range pc.Messages() {
-		// TODO: Do stuff with message.
+		var input api.InputRoomEvent
+		if err := json.Unmarshal(message.Value, &input); err != nil {
+			c.logError(message, err)
+		} else {
+			if err := processRoomEvent(c.DB, input); err != nil {
+				c.logError(message, err)
```

Comment:
You don't explain why you only log the error instead of something more drastic. Please explain.